js/plates-model/earthquake.js

```diff
@@ -0,0 +1,58 @@
+import { serialize, deserialize } from '../utils'
+import config from '../config'
+import { random } from '../seedrandom'
+
+export default class Earthquake {
+  static shouldCreateEarthquake (field) {
+    // There are two cases possible:
+    // A. Field is in the subduction zone. Then, the earthquake should be more likely to show up when the subduction
+    //    is progressing faster (relative speed between plates is higher)
+    if (field.subductingFieldUnderneath) {
+      if (random() < field.subductingFieldUnderneath.subduction.relativeVelocity.length() * config.earthquakeInSubductionZoneProbability) {
+        return true
+      }
+    }
+    // B. Field is next to the divergent boundary. Then, the earthquake should be more likely to show up when the
+    //    plate is moving faster and divergent boundary is more visible.
+    return field.divergentBoundaryZone && random() < field.linearVelocity.length() * config.earthquakeInDivergentZoneProbability
+  }
+
+  constructor (field) {
+    // Earthquake can be shallow or deep. Shallow are placed around top plate crust, while deep are placed
+    // where two plates are colliding (so we use colliding field elevation as a base).
+    const subductingField = field.colliding
+    let earthquakeElevation
+    if (subductingField) {
+      // Subduction zone
+      const deep = subductingField && random() < 0.5
+      earthquakeElevation = deep ? subductingField.elevation - 0.2 * random() : field.elevation - 0.3 * random()
```

Comment:
I'm not sure what the constants of 0.2 or 0.3 represent here

Reply:
It's range of the random depth part.
If the earthquake is shallow, it's placed in `[field.elevation, field.elevation - 0.3]` range.
If the earthquake is deep, it's placed in `[subductingField.elevation, subductingField.elevation - 0.2]` range (around area where two plates collide).

If you look at some comments and config, you should find information that elevation range is between 1 and -3.3 (-3.3 is defined by `config.subductionMinElevation`):
1 is the highest moubtain possible. 0 is the deepest trench. Anything below 0 is subducting plate. -3.3 is the min elevation of the subducting plate and then we remove field that went below this value:

```
  //  - [0, 1] -> [the deepest trench, the highest mountain]
  //  - 0.5 -> sea level
  //  - [config.subductionMinElevation, 0] -> subduction
```

So, 0.2 and 0.3 are some magic params based on this [0, -3.3] range. Most likely they will require tweaking when we can see earthquakes in the cross-section.

Reply:
this makes a lot more sense, thanks. I've made changes to the cross section so hopefully it should be a more accurate reflection of these values.